### PR TITLE
Copy dockerfile into source directory before staging

### LINKED ISSUE #74: For staging with dockerfile, dockerfile must be in exploded-war directory

### PR DESCRIPTION
**Pull request by loosebazooka**
fixes #74

### CODE REVIEW

**Comment by loosebazooka** [comment]
@meltsufin

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineStandardStaging.java [hunk 1]
```diff
@@ -68,16 +68,16 @@ public void stageStandard(StageStandardConfiguration config) throws AppEngineExc
...
     try {
-
-      cloudSdk.runAppCfgCommand(arguments);
-
       if (dockerfile != null && dockerfile.toFile().exists()) {
         Files.copy(dockerfile, dockerfileDestination, StandardCopyOption.REPLACE_EXISTING);
       }
 
+      cloudSdk.runAppCfgCommand(arguments);
+
```
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineStandardStaging.java]: nit: extra blank line(s) here

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineStandardStaging.java [hunk 2]
```diff
@@ -68,16 +68,16 @@ public void stageStandard(StageStandardConfiguration config) throws AppEngineExc
 
     Path dockerfile =
         config.getDockerfile() == null ? null : config.getDockerfile().toPath();
-    Path dockerfileDestination = config.getStagingDirectory().toPath();
+    Path dockerfileDestination = config.getSourceDirectory().toPath();
```
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineStandardStaging.java]: Maybe rename to `sourceDirectoryPath`

Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineStandardStaging.java]: that changes the whole config unfortunately.

Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineStandardStaging.java]: I just meant dockerfileDestination -> sourceDirectoryPath
Even better, just use config.getSourceDirectiory().toPath() directly in Files.copy statement.

Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineStandardStaging.java]: hah, I thought you were talking about the confusion between sourceDir and stagingDir....

**Comment by meltsufin** [comment]
:ship:

**Pull request merged**